californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -108,7 +108,7 @@
 	private boolean detachExecutor;
 
 	/** The endpoint. */
-	private Endpoint endpoint;
+	protected Endpoint endpoint;
```

Comment:
will be obsolete using `CoapStackFactory`.

Reply:
Okay. But this forces me to do the re-design in this PR. I have no issues with this change, but I thought you wanted it to happen in a new PR?

Reply:
OK, so if this will be part of the redesign and new PR, then I will just not review it here.

Reply:
Great.